ipamutils/utils.go

```diff
@@ -13,14 +14,29 @@ var (
 	// PredefinedGranularNetworks contains a list of 64K IPv4 private networks with host size 8
 	// (10.x.x.x/24) which do not overlap with the networks in `PredefinedBroadNetworks`
 	PredefinedGranularNetworks []*net.IPNet
-
-	initNetworksOnce sync.Once
+	initNetworksOnce           sync.Once
 )
 
-// InitNetworks initializes the pre-defined networks used by the built-in IP allocator
-func InitNetworks() {
+// PredefinedPools represent a set of address pools with prefix length Size.
+// Each pool in the set is derived from the Base pool. Base is to be passed
+// in CIDR format. The Scope field can be "local" or "global", and says
+// whether the address pool is to be used for local or global scope networks.
+// Example: a Base "10.10.0.0/16 with Size 24 will define the set of 256
+// 10.10.[0-255].0/24 address pools
+type PredefinedPools struct {
+	Base string `json:"base"`
+	Size int    `json:"size"`
+}
+
+// InitNetworks initializes the local predefined address pools
+// with the default values.
+func InitNetworks(defaultAddressPool []*PredefinedPools) {
 	initNetworksOnce.Do(func() {
-		PredefinedBroadNetworks = initBroadPredefinedNetworks()
+		if defaultAddressPool != nil {
+			InitAddressPools(defaultAddressPool)
```

Comment:
nit (at the risk of bike shedding here)...  I get that the InitAddressPools() code is more generic than just initializing BroadPredefinedNetworks in capability .... and yet it's only called for its side effect on said global variable.  I would recommend making InitAddressPools return ([]*net.IPNet, error) returning `nil, nil` on a nil parameter and changing the code to something like:
```go
        PredefinedBoradNetworks, err := InitAddressPools(defaultAddressPool)
        if err != nil {
                // log error, presumably to warn of a bad configuration
        } 
        if PredefinedBroadNetworks == nil {
                PredefinedBroadNetworks = initBroadPredefinedNetworks()
        }
```

As it is, the code is ignoring the error generated by InitAddressPools() which seems wrong.

Reply:
made code change to to check for error code. But retaining the API syntax with only one return value .